etc/scripts/release.hs

```diff
@@ -268,6 +268,20 @@ rules global@Global{..} args = do
                 cmd "strip -o"
                     [out, releaseBinDir </> binaryName </> stackExeFileName]
 
+    releaseDir </> binaryInstallerFileName %> \out -> do
+        need [releaseDir </> binaryExeFileName]
+        cmd "stack etc/scripts/build-stack-installer.hs" :: Action ()
```

Comment:
I'd suggest moving building the `.nsi` file to a separate target that's just responsible for running `build-stack-instasller.hs`.  Also make sure you `need` build-stack-installer.hs, so that if that file is changed it triggers re-running the script and generating the NSI.

Reply:
You mean something like this? https://github.com/commercialhaskell/stack/pull/5088/commits/493186304831fa5f19a58283a6692ff1342141c1#diff-cf9425a8bb87aa8706fdee9e97173e34R285